File: Easy/9. Palindrome Number/PalindromeNumber.java
/*
LeetCode Easy: 9. Palindrome Number
https://leetcode.com/problems/palindrome-number/

Check PalindromeNumber.cpp for brief solution explanation.
*/
class Solution {
    public boolean isPalindrome(int x) {
        String num = Integer.toString(x);
        int N = num.length();
        for(int i = 0;i<N/2;i++){
            if(num.charAt(i)!=num.charAt(N-i-1)){
                return false;
            }
        }
        return true;
    }
}

File: Easy/9. Palindrome Number/PalindromeNumber.py
'''
LeetCode Easy: 9. Palindrome Number
https://leetcode.com/problems/palindrome-number/

Check PalindromeNumber.cpp for brief solution explanation.
'''
class Solution:
    def isPalindrome(self, x: int) -> bool:
        num = str(x)
        N = len(num)
        for i in range(N//2):
            if num[i]!=num[N-i-1]:
                return False
        return True